File: bench/micro/loop_overhead.go
// Go equivalent of loop_overhead.ap.
// XOR accumulation defeats DCE without overflow concerns.
package main

import (
	"fmt"
	"os"
	"time"
)

func main() {
	iters := 100000000 + os.Getpid()
	t0 := time.Now()
	acc := os.Getpid()
	for i := 0; i < iters; i++ {
		acc = acc ^ i
	}
	elapsed := time.Since(t0).Nanoseconds()
	fmt.Printf("iters=%d\n", iters)
	fmt.Printf("acc=%d\n", acc)
	fmt.Printf("elapsed_ns=%d\n", elapsed)
}

File: bench/micro/loop_overhead.py
"""Python equivalent of loop_overhead.ap.
XOR accumulation — no bignum blowup since XOR of two small ints
stays small.
"""

import os
import time

iters = 100_000_000 + os.getpid()
t0 = time.monotonic_ns()
acc = os.getpid()
for i in range(iters):
    acc = acc ^ i
elapsed = time.monotonic_ns() - t0
print(f"iters={iters}")
print(f"acc={acc}")
print(f"elapsed_ns={elapsed}")
